### ISSUE TITLE
rustc(4556,0x7fffcca583c0) malloc: *** malloc_zone_unregister() failed for 0x7fffcca4f548 on OSX 10.12 beta 2

### ISSUE THREAD

**Issue by glandium** [rust-lang/rust#34674]
Heads up: Running the rust compiler on OSX 10.12 beta 2 leads to the following error message:
> rustc(4556,0x7fffcca583c0) malloc: *** malloc_zone_unregister() failed for 0x7fffcca4f548

Then the process gets stuck.

This roots from the malloc zone registration code in jemalloc 3, which is shared with Firefox's replace-malloc and causes the same issue on Firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1284677)

Filed jemalloc issue: https://github.com/jemalloc/jemalloc/issues/420

**Comment by brson** [comment]
Thanks for the report! This is scheduled for release Thursday. @glandium is 10.12 beta 2 compatible with stable and/or nightly?

**Comment by brson** [comment]
Since this is an unreleased version of OS X, and the final release isn't scheduled until fall, I think it's not a pressing issue, but we should solve it for 1.11.

**Comment by brson** [comment]
Nominating because this is a high priority issue someone should investigate.

**Comment by e-oz** [comment]
> I think it's not a pressing issue, but we should solve it for 1.11.

cool, I can't use Sierra because I use Rust for work...

**Comment by jansegre** [comment]
As a data point, I don't experience this issue on nightly (9f1ffdd 2016-07-11), but I do on stable. I'm on macOS 10.12 Beta (16A238m).

**Comment by tzakharko** [comment]
I am still experiencing this issue when trying to *build* rust on OS X.  Rustc just enters a (seemingly) infinite loop with a comparable error message. Note: this also happens if I disable jemalloc via the --disable-jemalloc option to configure. This is under Sierra Dev build 6 (16A294a), Rust 413ada30402f823a86a75578afc63d07179f6a23

**Comment by e-oz** [comment]
@tzakharko can't reproduce.
Beta 12 installed successfully and working fine on Sierra 10.12 Beta (16A294a).

**Comment by tzakharko** [comment]
@e-oz True, Beta 12 is building and working correctly. However, the current master does not. I do not know whether its a regression or not, as I have no idea about how Rust repo is organised.

**Comment by eshansingh** [comment]
Hey, just wanted to know what's the status on this? jemalloc has fixed this on their end, by PR https://github.com/jemalloc/jemalloc/pull/427. Is there an update yet for this?

**Comment by zeusdeux** [comment]
This is still broken on macOS Sierra version 10.12 (16A323).

I tried to upgrade my local rust installation using homebrew (`brew update && brew upgrade rust`) and that lead to the build hanging on `make`

Output reported by homebrew:

```
==> ./configure --prefix=/usr/local/Cellar/rust/1.11.0 --enable-clang --release-channel=stable
==> make
```

**Comment by tzakharko** [comment]
@zeusdeux Installing with --HEAD works

**Comment by cielavenir** [comment]
However currently we have to kill docdir stuff from install.mk.
https://raw.githubusercontent.com/cielavenir/homebrew-ciel/master/patch/rust.patch

**Comment by shanemikel** [comment]
I get this failure during the configure step on Sierra.  This is happening in the current macports package.

**Comment by keeganstothert** [comment]
came issue on official release of high sierra. rustc errors (with above error) and cargo just hangs.

**Comment by peterbe** [comment]
Still a thing apparently. I'm on macOS 11.6.1 and I had (long ago) installed rust with `brew`. I'm pretty sure it's the `brew` upgrade here. I ran `brew upgrade rust` 
Before i had:
```
▶ rustc --version
rustc 1.56.0
```

The upgrade looked like this:
<img width="563" alt="Screen Shot 2021-12-18 at 2 35 33 PM" src="https://user-images.githubusercontent.com/26739/146653644-088bc276-6d97-4000-bdb6-a5fb00fa7237.png">

I've lost it now but I got the this same error (OT of this issue) when running `rustc --version`. 

Anyway, *think* what happened was that I had started a terminal BEFORE the upgrade. Creating a fresh new terminal now "solves" it and all is well:

```
▶ rustc --version
rustc 1.56.1
```

So basically, if anybody else sees this: **create a new terminal** and see if that makes the error go away.
